cpp/src/arrow/flight/internal.cc

```diff
@@ -105,7 +108,15 @@ static Status FromGrpcContext(const grpc::ClientContext& ctx, Status* status) {
     message += ". Detail: ";
     message += std::string(detail_ref.data(), detail_ref.size());
   }
-  *status = Status(code, message);
+  const auto grpc_detail_val = trailers.find(kBinaryErrorDetailsKey);
+  std::shared_ptr<FlightStatusDetail> flightStatusDetail(
+      new FlightStatusDetail(FlightStatusCode::Unavailable));
```

Comment:
We shouldn't use Unavailable. We should probably add a new enum variant for when there's no other applicable code, or otherwise default to INTERNAL.

Reply:
fixed in most recent path